core/src/main/java/org/apache/iceberg/io/FileAppenderFactory.java

```diff
@@ -19,14 +19,15 @@
 
 package org.apache.iceberg.io;
 
+import java.io.Serializable;
 import org.apache.iceberg.FileFormat;
 
 /**
  * Factory to create a new {@link FileAppender} to write records.
  *
  * @param <T> data type of the rows to append.
  */
-public interface FileAppenderFactory<T> {
+public interface FileAppenderFactory<T> extends Serializable {
```

Comment:
This isn't required for Spark. Could we add this to the `FlinkFileAppenderFactory` instead?

Reply:
That sounds reasonable.